Tasks/Inc/RTOS_BPS.h

```diff
@@ -2,55 +2,121 @@
+
+/**
+ * @brief Posts a semaphore
+ * @param sem4 this is a semaphore pointer to post to
+ * @param opt determines the type of POST performed
+ * @return The current value of the semaphore counter or 0 upon error 
+ */
+BPS_OS_SEM_CTR RTOS_BPS_SemPost(BPS_OS_SEM *sem4, BPS_OS_OPT opt);
+
+/*
+ * @brief Initializes a mutex object.
+ * @param *mut - pointer to a mutex to initialize
+ * @param name - char* of the name of the mutex
+ * @return none
+ */
+void RTOS_BPS_MutexCreate(BPS_OS_MUTEX *mut, char* name);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief   Function waits for Mutex
+ * @param   *mutex - pointer to mutex
+ * @param   options - determines what the mutex will do, ie: block or not block
  */
-void RTOS_BPS_SemPend(void);
+void RTOS_BPS_MutexPend(BPS_OS_MUTEX* mutex, BPS_OS_OPT opt);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief   Posts the specified Mutex
+ * @param   *mutex - pointer to the specified RTOS Mutex object
+ * @param   options - a parameter which determines what kind of Post MutexPost performs
+ * @return  none
  */
-void RTOS_BPS_SemPost(void);
+void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief   Creates a task that will be handled by the OS
+ * @param   *p_tcb - pointer to the tcb
+ * @param   *p_name - pointer to task name
+ * @param   p_task - pointer to the task
+ * @param   *p_args - pointer to task function arguments
+ * @param   *prio - task priority
+ * @param   *p_stk_base - the stack
+ * @param   stk_size - size of the stack
+ * @param   *p_err - return error code
+ * @return  nothing to see here
  */
-void RTOS_BPS_MutexPend(void);
+void RTOS_BPS_TaskCreate(
+    BPS_OS_TCB      *p_tcb,
+    char         *p_name,
+    void         *p_task,
+    void         *p_arg,
+    uint8_t       prio,
+    BPS_CPU_STK  *p_stk_base,
+    uint64_t      stk_size
+);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief Creates a Second-based Time Delay.
+ * @param dly Defines how many seconds to delay for.
+ * @return none 
  */
-void RTOS_BPS_MutexPost(void);
+void RTOS_BPS_DelaySecs(uint16_t dly);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief Creates a Millisecond-based Time Delay.
+ * @param dly Defines how many milliseconds to delay for.
+ * @return none 
  */
-void RTOS_BPS_TaskCreate(void);
+void RTOS_BPS_DelayMs(uint16_t dly);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief Creates a semaphore with the initially specified count
+ * 
+ * @param sem - pointer to a semaphore object to create and initialize
+ * @param name - name of the semaphore
+ * @param count - initial count for the semaphore
  */
-void RTOS_BPS_TimeDelay(void);
+void RTOS_BPS_SemCreate(BPS_OS_SEM* sem, char* name, uint32_t count);
 
+/**
+ * @brief Creates a Tick-based Time Delay.
+ * @param dly Defines how many ticks to delay for.
+ * @return none 
+ */
+void RTOS_BPS_DelayTick(BPS_OS_TICK dly);
```

Comment:
Reorder the functions in this file to be more structured (sem functions next to each other, etc.)